src/app/api/toy-example/README.md

```diff
@@ -0,0 +1,398 @@
+          message: "The email is invalid",
+        }),
+        { status: 400 }
+      );
+    } else if (error.password) {
+      return NextResponse.json(
+        SignInResponse.parse({
+          code: "INVALID_PASSWORD",
+          message: "The password is invalid",
+        }),
+        { status: 400 }
+      );
+    } else {
+      // When we add new fields to the schema, we may forget to handle it
+      // in the API route. This last case serves a catch-all.
+      return NextResponse.json(
+        SignInResponse.parse({
+          code: "UNKNOWN",
+          data: "Forgot to handle an error",
+        }),
+        { status: 500 }
+      );
+    }
+  } else {
+    // Successful parse...
+  }
+};
+```
+
+If the request body is successfully parsed, we can check whether the email is unique.
+
+```ts
+// route.ts
+
+// Import statements...
+// Database
+
+export const POST = async (request: NextRequest) => {
+  // safeParse prevents an error from being thrown.
+  const maybeBody = SignInRequest.safeParse(await request.json());
+
+  if (!maybeBody.success) {
+    // Check which field fails the schema...
+  } else {
+    const body = maybeBody.data;
+
+    // Checks if email is unique.
+    const hasSameEmail = DATABASE.some((user) => user.email === body.email);
+    if (hasSameEmail) {
+      return NextResponse.json(
+        SignInResponse.parse({
+          code: "EMAIL_EXIST",
+          message: "The email existed already",
+        }),
+        { status: 400 }
+      );
+    }
+  }
+};
+```
+
+If the new user's email is unique, we can safely add it to our database and returns a success status.
+
+```ts
+// route.ts
+
+// Import statements...
+// Database
+
+export const POST = async (request: NextRequest) => {
+  // safeParse prevents an error from being thrown.
+  const maybeBody = SignInRequest.safeParse(await request.json());
+
+  if (!maybeBody.success) {
+    // Check which field fails the schema...
+  } else {
+    const body = maybeBody.data;
+
+    // Checks if email is unique.
+    const hasSameEmail = DATABASE.some((user) => user.email === body.email);
+    if (hasSameEmail) {
+      // ...
+    }
+
+    // Add user to database and return success.
+    DATABASE.push(body);
+    return NextResponse.json(
+      SignInResponse.parse({
+        code: "SUCCESS",
+        message: "The new user was successfully registered",
+      }),
+      { status: 200 }
+    );
+  }
+};
+```
+
+That's it. The final step is to test your API route - either through the web app or [Postman](https://www.postman.com/). Congrats on making it here 😁! I hope you've learned something new and feel somewhat more prepared for backend development.
+
+### Afterwords
+
+The tutorial serves as a high-level overview for backend development. If you're interested in learning more, I recommend exploring the following topics:
+
+1. [Cache](https://nextjs.org/docs/app/building-your-application/caching#data-cache) to reduce the overhead of making network requests,
```

Comment:
I would be a bit worried that giving people Next.js's docs on Cache and Middleware right now might be a bit overwhelming... especially caching is kinda complicated in Next.js 13. 

Maybe instead, they could take a look at building pages in React + Next.js next? (Or keep both)

Reply:
You're right, I can just completely remove this section.